server/webapp/WEB-INF/rails.new/app/controllers/api_v4/agents_controller.rb

```diff
@@ -0,0 +1,129 @@
+        render_http_operation_result(result, { data: json })
+      end
+    end
+
+    def destroy
+      result = HttpOperationResult.new
+      agent_service.deleteAgents(current_user, result, [params[:uuid]])
+      render_http_operation_result(result)
+    end
+
+    def bulk_update
+      result                = HttpLocalizedOperationResult.new
+      uuids                 = params[:uuids]
+      resources_to_add      = params[:operations][:resources][:add]
+      resources_to_remove   = params[:operations][:resources][:remove]
+      environments_to_add   = params[:operations][:environments][:add]
+      environment_to_remove = params[:operations][:environments][:remove]
+      agent_service.bulkUpdateAgentAttributes(current_user, result, uuids, resources_to_add, resources_to_remove, environments_to_add, environment_to_remove, to_enabled_tristate)
+      render_http_operation_result(result)
+    end
+
+    def bulk_destroy
+      result = HttpOperationResult.new
+      agent_service.deleteAgents(current_user, result, Array.wrap(params[:uuids]))
+      render_http_operation_result(result)
+    end
+
+    private
+
+    def set_default_values_if_not_present
+      params[:uuids]                              = params[:uuids] || []
+      params[:operations]                         = params[:operations] || {}
+      params[:operations][:resources]             = params[:operations][:resources] || {}
+      params[:operations][:environments]          = params[:operations][:environments] || {}
+      params[:operations][:resources][:add]       = params[:operations][:resources][:add] || []
+      params[:operations][:resources][:remove]    = params[:operations][:resources][:remove] || []
+      params[:operations][:environments][:add]    = params[:operations][:environments][:add] || []
+      params[:operations][:environments][:remove] = params[:operations][:environments][:remove] || []
+    end
+
+    def maybe_join(obj)
+      if obj.is_a?(Array)
+        obj.join(',')
+      else
+        obj
+      end
+    end
+
+    def set_cache_control
+      response.headers['Cache-Control'] = 'private, must-revalidate'
+    end
+
+    private
+
+    attr_reader :agent_instance
+
+    def to_enabled_tristate
+      enabled = params[:agent_config_state]
+      if enabled.blank?
+        TriState.UNSET
+      elsif enabled =~ /\Aenabled\Z/i
+        TriState.TRUE
+      elsif enabled =~ /\Adisabled\Z/i
+        TriState.FALSE
+      else
+        raise BadRequest.new('The value of `agent_config_state` can be one of `Enabled`, `Disabled` or null.')
+      end
+    end
+
+    def load_agent
+      @agent_instance = agent_service.findAgent(params[:uuid])
+      raise RecordNotFound if @agent_instance.nil? || @agent_instance.isNullAgent()
+    end
+
+    def agent_presenter
+      ApiV4::AgentRepresenter.new([@agent_instance, environment_config_service.environmentsFor(@agent_instance.getUuid()), security_service, current_user])
+    end
```

Comment:
Should we consider using a hash instead of an array for this, because we rely on the order of the parameters in the representer. Changing the order might lead to inconsistent behaviour.